bin/custom/concerttours/testsrc/concerttours/service/impl/DefaultBandServiceUnitTest.java

```diff
@@ -0,0 +1,71 @@
+package concerttours.service.impl;
+
+import concerttours.daos.BandDAO;
+import concerttours.model.BandModel;
+import de.hybris.bootstrap.annotations.UnitTest;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Collections;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@UnitTest
+public class DefaultBandServiceUnitTest {
+    /**
+     * Name of test band.
+     */
+    private static final String BAND_CODE = "Ch00X";
+    /**
+     * Name of test band.
+     */
+    private static final String BAND_NAME = "Singers All";
+    /**
+     * History of test band.
+     */
+    private static final String BAND_HISTORY =
+            "Medieval choir formed in 2001, based in Munich famous for authentic monastic chants";
+    private DefaultBandService bandService;
+    private BandDAO bandDAO;
+    private BandModel bandModel;
+
+    @Before
+    public void setUp() {
+        bandService = new DefaultBandService();
+
+        bandDAO = mock(BandDAO.class);
+        bandService.setBandDAO(bandDAO);
+
+        bandModel = new BandModel();
+        bandModel.setCode(BAND_CODE);
+        bandModel.setName(BAND_NAME);
+        bandModel.setHistory(BAND_HISTORY);
+    }
+
+    /**
+     * This test tests and demonstrates that the Service's getAllBands method calls the DAOs' getBands method
+     * and returns the data it receives from it.
+     */
+    @Test
+    public void getAllBands_existingBand_band() {
+        final List<BandModel> bandModels = Collections.singletonList(bandModel);
```

Comment:
code inside test methods should be separated into blocks (by newlines) - for example
1st block - creating instances
2nd block - mocking instance behaviour (when...then...)
3rd block - test method invocation
4th block - asserts

+ if you can move the items into the global variables - fo it